src/SongManager.cpp

```diff
@@ -356,6 +356,32 @@ void SongManager::AddGroup( RString sDir, RString sGroupDirName, Group* group )
 	//m_sSongGroupBackgroundPaths.push_back( sBackgroundPath );
 }
 
+
+void SongManager::RemoveGroup( RString sDir, RString sGroupDirName, Group* group )
+{
+	unsigned j;
+	for(j = 0; j < m_sSongGroupNames.size(); ++j)
+		if( sGroupDirName == m_sSongGroupNames[j] ) {
+			break; 
+		}
+
+	if ( group == nullptr ) {
+		LOG->Warn( "Could not RemoveGroup '%s'. Group object is null.", sGroupDirName.c_str() );
+		return;
+	}
+
+	// Remove the group from its series if the group has one and if the series exists
+	if( group->GetSeries() != "" )
+	{
+		std::unordered_set<Group*>& series = m_mapSeriesToGroups[group->GetSeries()];
+		series.erase(group);
+	}
+
+	m_sSongGroupBannerPaths.erase(m_sSongGroupBannerPaths.begin() + j);

```

Comment:
What if the sGroupDirName == m_sSongGroupNames[j] check was never met. Then j would point outside of the size. Add validation checks.